elide-core/src/main/java/com/yahoo/elide/core/pagination/Pagination.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core.pagination;
+
+import com.yahoo.elide.core.exceptions.InvalidValueException;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.ToString;
+
+import javax.ws.rs.core.MultivaluedMap;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Encapsulates the pagination strategy.
+ */
+
+@AllArgsConstructor
+@ToString
+public class Pagination {
+
+    /**
+     * Denotes the internal field names for paging.
+     */
+    public enum PaginationKey { offset, limit }
+
+    public static final int DEFAULT_PAGE_SIZE = 500;
+    public static final int MAX_PAGE_SIZE = 10000;
+
+    private static final Pagination DEFAULT_PAGINATION = new Pagination(0, 0);
+
+    public static final Map<String, PaginationKey> PAGE_KEYS = new HashMap<>();
+    static {
+        PAGE_KEYS.put("page[size]", PaginationKey.limit);
+        PAGE_KEYS.put("page[limit]", PaginationKey.limit);
+        PAGE_KEYS.put("page[number]", PaginationKey.offset);
+        PAGE_KEYS.put("page[offset]", PaginationKey.offset);
+    }
+
+
+    private int offset;
+
+    @Getter
+    private int limit;
+
+    public int getOffset() {
+        return (offset > 0 ? offset - 1 : 0) * limit;
```

Comment:
Maybe I'm missing something, but this still doesn't seem right.  Here is an example:
page=2, size=50 I believe is the same things as offset=100, limit=50.
So if someone provides and offset of 100, I wouldn't multiply that 50 (but I would if they provided page).

Does this make sense?  Am I missing something here?  I feel like the multiplication has to happen depending on whether or not they used 'offset' or 'page' in the query parameter.